controllers/toolchainconfig/toolchainconfig_controller.go

```diff
@@ -60,12 +73,70 @@ func (r *Reconciler) Reconcile(request reconcile.Request) (reconcile.Result, err
 			// Request object not found, could have been deleted after reconcile request.
 			// Owned objects are automatically garbage collected. For additional cleanup logic use finalizers.
 			// Return and don't requeue
-			reqLogger.Error(err, "it looks like the ToolchainConfig resource with the name 'config' was removed - the cache will use the latest version of the resource")
+			reqLogger.Error(err, "it looks like the toolchainconfig resource was removed - the cache will use the latest version of the resource")
 			return reconcile.Result{}, nil
 		}
 		// Error reading the object - requeue the request.
-		return reconcile.Result{}, err
+		reqLogger.Error(err, "error getting the toolchainconfig resource")
+		return defaultReconcile, err
 	}
 	updateConfig(toolchainConfig)
-	return reconcile.Result{}, nil
+
+	// Sync member configs to member clusters
+	sync := synchronizer{
+		logger:         reqLogger,
+		getMembersFunc: r.GetMembersFunc,
+	}
+
+	if syncErrs := sync.syncMemberConfigs(toolchainConfig); len(syncErrs) > 0 {
+		for cluster, errMsg := range syncErrs {
+			err := fmt.Errorf(errMsg)
+			reqLogger.Error(err, "error syncing to member cluster '%s'", cluster)
+		}
+		return defaultReconcile, r.setSyncFailureStatusCondition(reqLogger, toolchainConfig, syncErrs)
+	}
+	return defaultReconcile, r.setCompleteStatusCondition(reqLogger, toolchainConfig)
+}
+
+// setSyncFailureStatusCondition sets the ToolchainConfig status condition to `SyncComplete=false/reason=syncFailure`
+func (r *Reconciler) setSyncFailureStatusCondition(reqLogger logr.Logger, toolchainConfig *toolchainv1alpha1.ToolchainConfig, syncErrs map[string]string) error {
+	toolchainConfig.Status.SyncErrors = syncErrs
+	toolchainConfig.Status.Conditions = []toolchainv1alpha1.Condition{ToSyncFailure()}
+	err := r.Client.Status().Update(context.TODO(), toolchainConfig)
```

Comment:
Why not using the function from toolchain-common?
```
AddOrUpdateStatusConditionsWithLastUpdatedTimestamp
```

Reply:
Updated the status updates to use a common function that makes use of the AddOrUpdateStatusConditions. See 83a653b2b5db2f88a73a770838fac9536c6a3ffd

Reply:
btw, I proposed to use `AddOrUpdateStatusConditionsWithLastUpdatedTimestamp` because that one also updates the `LastUpdatedTime`.
`AddOrUpdateStatusConditions` only compares the content of the condition but not timestamps, so if there is no change, then the condition is not updated.